lib/libxdp/xsk.c

```diff
@@ -1072,13 +1072,17 @@ struct xsk_socket *xsk_socket__create_opts(const char *ifname,
 	}
 	rx = OPTS_GET(opts, rx, NULL);
 	tx = OPTS_GET(opts, tx, NULL);
-	fill = OPTS_GET(opts, fill, umem->fill_save);
-	comp = OPTS_GET(opts, comp, umem->comp_save);
+	fill = OPTS_GET(opts, fill, NULL);
+	comp = OPTS_GET(opts, comp, NULL);
 
 	if (!umem || !(rx || tx)) {
 		err = -EFAULT;
 		goto err;
 	}
+	if (!fill && !comp) {
```

Comment:
Probably no, because a `umem` must have both `fill` and `comp` rings to transfer ownership of UMEM frames between the kernel and the user-space application[0]. Thus, it is invalid that only one of `fill` and `comp` are set.

[0] https://docs.kernel.org/networking/af_xdp.html